plugins/scrollmsg.sma

```diff
@@ -14,94 +14,136 @@
 #include <amxmodx>
 #include <amxmisc>
 
-#define SPEED 0.3
-#define SCROLLMSG_SIZE	512
+const SCROLLMSG_SIZE = 512;
+const SCROLLMSG_TASK = 123;
 
-new g_startPos
-new g_endPos
-new g_scrollMsg[SCROLLMSG_SIZE]
-new g_displayMsg[SCROLLMSG_SIZE]
-new Float:g_xPos
-new g_Length
-new g_Frequency
+new g_hostname[64];
+new g_amx_scrollmsg_color_red;
+new g_amx_scrollmsg_color_green;
+new g_amx_scrollmsg_color_blue;
+new g_amx_scrollmsg_only_dead;
+new Float:g_amx_scrollmsg_speed;
+new Float:g_amx_scrollmsg_x_move_amount;
+new Float:g_amx_scrollmsg_x_start_pos;
+new Float:g_amx_scrollmsg_x_end_pos;
+new Float:g_amx_scrollmsg_y_pos;
+
+new g_start_pos;
+new g_end_pos;
+new g_scroll_msg[SCROLLMSG_SIZE];
+new g_display_msg[SCROLLMSG_SIZE];
+new g_length;
+new g_frequency;
+new g_hud_object;
+new Float:g_x_pos;
 
 public plugin_init()
 {
-	register_plugin("Scrolling Message", AMXX_VERSION_STR, "AMXX Dev Team")
-	register_dictionary("scrollmsg.txt")
-	register_dictionary("common.txt")
-	register_srvcmd("amx_scrollmsg", "setMessage")
+	register_plugin("Scrolling Message", AMXX_VERSION_STR, "AMXX Dev Team");
+
+	register_dictionary("scrollmsg.txt");
+	register_dictionary("common.txt");
+
+	register_srvcmd("amx_scrollmsg", "setMessage", _, "<message> <frequency>");
+	bind_pcvar_string(get_cvar_pointer("hostname"), g_hostname, charsmax(g_hostname));
+
+	bind_pcvar_num(create_cvar(   "amx_scrollmsg_color_red",       "200",      _, "Red color amount",                             true, 0.0, true, 255.0), g_amx_scrollmsg_color_red);
```

Comment:
How about to implode it to one cvar `amx_scrollmsg_color 200 100 0`?

Reply:
Not sure really, both seem fine to me. Let's hear more opinions.